distributed/semaphore.py

```diff
@@ -263,25 +322,36 @@ class Semaphore:
     """
 
     def __init__(self, max_leases=1, name=None, client=None):
-        # NOTE: the `id` of the `Semaphore` instance will always be unique, even among different
-        # instances for the same resource. The actual attribute that identifies a specific resource is `name`,
-        # which will be the same for all instances of this class which limit the same resource.
         self.client = client or get_client()
-        self.id = uuid.uuid4().hex
         self.name = name or "semaphore-" + uuid.uuid4().hex
         self.max_leases = max_leases
+        self.id = uuid.uuid4().hex
+        self._leases = deque()
 
-        if self.client.asynchronous:
-            self._started = self.client.scheduler.semaphore_create(
-                name=self.name, max_leases=max_leases
-            )
-        else:
-            self.client.sync(
-                self.client.scheduler.semaphore_create,
-                name=self.name,
-                max_leases=max_leases,
+        self._started = self.client.sync(
+            self.client.scheduler.semaphore_create,
+            name=self.name,
+            max_leases=max_leases,
+        )
+        # this should give ample time to refresh without introducing another
+        # config parameter since this *must* be smaller than the timeout anyhow
+        refresh_leases_interval = (
+            parse_timedelta(
+                dask.config.get("distributed.scheduler.locks.lease-timeout"),
+                default="s",
             )
-            self._started = asyncio.sleep(0)
+            / 5
+        )
+        self._refreshing_leases = False
+        pc = PeriodicCallback(
+            self._refresh_leases,
+            callback_time=1000 * refresh_leases_interval,
+            io_loop=self.client.io_loop,
+        )
+        self.refresh_callback = pc
+        # Registering here is important for proper cleanup
+        self.client._periodic_callbacks[f"refresh_semaphores_{self.id}"] = pc
```

Comment:
np: how about creating an attribute which has a value of `f"refresh_semaphores_{self.id}"` as it is used later on?